watertap/unit_models/electrodialysis_0D.py

```diff
@@ -518,10 +629,68 @@ def build(self):
                 initialize=500,
                 bounds=(0, 1000),
                 units=pyunits.amp * pyunits.meter**-2,
-                doc="Limiting Current Density accross the membrane as a function of the normalized length",
+                doc="Limiting current density across the membrane as a function of the normalized length",
             )
             self._make_performance_dl_polarization()
+        if (
+            not self.config.pressure_drop_method == PressureDropMethod.none
+        ) and self.config.has_pressure_change:
+            self._pressure_drop_calculation()
+
+            @self.Constraint(
+                self.flowsheet().time,
+                doc="Pressure drop expression as calculated by the pressure drop data, diluate.",
+            )
+            def eq_deltaP_diluate(self, t):
+                return self.diluate.deltaP[t] == -self.pressure_drop_total[t]
+
+            @self.Constraint(
+                self.flowsheet().time,
+                doc="Pressure drop expression as calculated by the pressure drop data,  concentrate.",
+            )
+            def eq_deltaP_concentrate(self, t):
+                return (
+                    # self.concentrate.deltaP[t]
+                    # == -self.pressure_drop[t] * self.cell_length
+                    self.concentrate.deltaP[t]
+                    == -self.pressure_drop_total[t]
+                )
+
+        elif self.config.pressure_drop_method == PressureDropMethod.none and (
+            not self.config.has_pressure_change
+        ):
+            pass
+        else:
+            raise ConfigurationError(
+                "A valid (not none) pressure_drop_method and has_pressure_change being True "
+                "must be both used or unused at the same time. "
+            )
+
         # Build Constraints
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Calculate flow velocity in a single diluate channel",
```

Comment:
Probably doesn't matter too much, but it may be worth including in the doc string that the velocity is based on the arithmetic average of volumetric flow rate (opposed to inlet velocity, outlet velocity, or other forms of averaging) because of the potential significance of volume change due to water permeation. It is a slight nuance where the 1D is calculated with the velocity in each discrete modeling element.

Reply:
I added a line saying that "based on the average of the inlet and outlet."